File: app/src/main/java/com/my/televip/virtuals/messenger/UserConfig.java
package com.my.televip.virtuals.messenger;

import com.my.televip.Class.ClassLoad;
import com.my.televip.Class.ClassNames;
import com.my.televip.obfuscate.Obfuscate;
import com.my.televip.virtuals.tgnet.TLRPC;

import de.robv.android.xposed.XposedHelpers;

public class UserConfig {

    Object userConfig;

    public UserConfig(Object obl){
        userConfig = obl;
    }

    public static int getSelectedAccount() {
        String selectedAccountField = Obfuscate.getFieldName("UserConfig", "selectedAccount");
        return XposedHelpers.getStaticIntField(ClassLoad.getClass(ClassNames.USER_CONFIG), selectedAccountField);
    }

    public long getClientUserId(){
        return XposedHelpers.getLongField(userConfig, Obfuscate.getFieldName("UserConfig" , "clientUserId"));
    }

    public TLRPC.User getCurrentUser(){
        return new TLRPC.User(XposedHelpers.callMethod(userConfig, Obfuscate.getMethodName("UserConfig" , "getCurrentUser")));
    }

    public static UserConfig getInstance(int num) {
        return new UserConfig(XposedHelpers.callStaticMethod(ClassLoad.getClass(ClassNames.USER_CONFIG), Obfuscate.getMethodName("UserConfig", "getInstance"), num));
    }

    public static UserConfig getUserConfig() {
        return getInstance(getSelectedAccount());
    }

    public Object get_UserConfig() {
        return userConfig;
    }

}

File: app/src/main/java/com/my/televip/virtuals/messenger/UserConfig.kt
package com.my.televip.virtuals.messenger

import com.my.televip.Class.ClassLoad
import com.my.televip.Class.ClassNames
import com.my.televip.obfuscate.Obfuscate
import com.my.televip.virtuals.tgnet.TLRPC
import de.robv.android.xposed.XposedHelpers

class UserConfig(@JvmField val userConfig: Any?) {

    val currentUser: TLRPC.User
        get() = TLRPC.User(XposedHelpers.callMethod(userConfig, Obfuscate.getMethodName("UserConfig", "getCurrentUser")))

    val clientUserId: Long
        get() = XposedHelpers.callMethod(userConfig, Obfuscate.getMethodName("UserConfig", "getClientUserId")) as? Long ?: 0L

    companion object {
        @JvmStatic
        fun getSelectedAccount(): Int {
            val userConfigClass = ClassLoad.getClass(ClassNames.USER_CONFIG) ?: return 0
            return XposedHelpers.getStaticIntField(userConfigClass, Obfuscate.getFieldName("UserConfig", "selectedAccount"))
        }
    }
}
